torch/csrc/jit/codegen/cuda/executor_launch_params.h

```diff
@@ -0,0 +1,91 @@
+
+#include <torch/csrc/jit/codegen/cuda/type.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace cuda {
+
+class TORCH_CUDA_API LaunchParams {
+ public:
+  unsigned int smem() const {
+    return smem_;
+  }
+  unsigned int nBlocks() const {
+    return gdimx_ * gdimy_ * gdimz_;
+  }
+
+  unsigned int nThreads() const {
+    return bdimx_ * bdimy_ * bdimz_;
+  }
+
+  unsigned int bdimx() const {
+    return (unsigned int)bdimx_ == -1 ? 1 : bdimx_;
+  }
+
+  unsigned int gdimx() const {
+    return (unsigned int)gdimx_ == -1 ? 1 : gdimx_;
+  }
+
+  unsigned int bdimy() const {
+    return (unsigned int)bdimy_ == -1 ? 1 : bdimy_;
+  }
+
+  unsigned int gdimy() const {
+    return (unsigned int)gdimy_ == -1 ? 1 : gdimy_;
+  }
+
+  unsigned int bdimz() const {
+    return (unsigned int)bdimz_ == -1 ? 1 : bdimz_;
+  }
+
+  unsigned int gdimz() const {
+    return (unsigned int)gdimz_ == -1 ? 1 : gdimz_;
+  }
+
+  void checkAndSet(
+      const int64_t incoming_val,
+      int& class_val,
+      std::string val) {
+    TORCH_INTERNAL_ASSERT(
+        class_val == -1 || incoming_val == 1 || class_val == 1 ||
+            incoming_val == class_val,
+        "Tried to set ",
+        val,
+        " to ",
+        incoming_val,
+        ", but it was already set and new value does not match.",
+        " Thread dims all have to be bound to the same value.");
+    TORCH_CHECK(
+        incoming_val > 0,
+        "Received a thread binding on ",
+        val,
+        " that is ",
+        incoming_val,
+        ". Cannot create negative threads.");
+    if (class_val == -1 || class_val == 1) {
```

Comment:
what is the meaning of -1 vs 1 ? maybe create symbolic constants for them?

Reply:
Added a todo to convert them to c10::optional. Though even that seems like a relatively heavy hammer.

Reply:
Using -1 and 1 as special values may be ok, but they could use good symbolic names

Reply:
NITPICK: Is `-1` even necessary?
In cases where we access un-initialized binding, we map `-1` to 1 anyway. So maybe we can just stick with 1 to begin with as default binding and not use c10::optional?